src/online2bin/online2-tcp-nnet3-decode-faster.cc

```diff
@@ -307,8 +307,9 @@ int main(int argc, char *argv[]) {
 
             decoder.FinalizeDecoding();
 
-            server.SaveRemainder(
-                samp_pipeline - feature_pipeline.NumFramesReady() * feature_pipeline.FrameShiftInSeconds() * samp_freq);
+            int32 processed = feature_pipeline.NumFramesReady() * feature_pipeline.FrameShiftInSeconds() * samp_freq;
+            if (processed < samp_pipeline)
```

Comment:
We also need to make sure `samp_pipeline - processed <= chunk_len`. That's why I suggested clipping. Otherwise the `offset` variable in the `SaveRemainder` method might become negative and `samp_buf_[i]` might be out of bounds.